Corgibytes.Freshli.Cli/Functionality/Analysis/CacheWasNotPreparedEvent.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using Corgibytes.Freshli.Cli.Functionality.Engine;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Corgibytes.Freshli.Cli.Functionality.Analysis;
+
+public class CacheWasNotPreparedEvent : ErrorEvent
+{
+    public IServiceProvider ServiceProvider { get; init; } = null!;
+
+    public string CacheDirectory { get; init; } = null!;
+    public string RepositoryUrl { get; init; } = null!;
+    public string? RepositoryBranch { get; init; }
+    public string HistoryInterval { get; init; } = null!;
+
+    public override void Handle(IApplicationActivityEngine eventClient) =>
+        // TODO: Actually prepare the cache before restarting the analysis
+        eventClient.Dispatch(new RestartAnalysisActivity(
```

Comment:
I was wondering if an event should have a 'side effect' like dispatching a new activity, or should this be done in an event listener?

Reply:
We chatted about this during a recent ensemble session, but I wanted to record a note of that conversation here. I think that this side effect is okay because we limit what the event can do. An event's `Handle` method can't directly fire other events, it can only dispatch activities. And similarly, for activities, an activity can't directly dispatch an activity from its `Handle` method, it's only able to fire events.